drivers/usb/bc12/bc12_pi3usb9201.c

```diff
@@ -0,0 +1,429 @@
+/*
+ * Copyright (c) 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* PI3USB9201 USB BC 1.2 Charger Detector driver. */
+
+#define DT_DRV_COMPAT diodes_pi3usb9201
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/usb/usb_bc12.h>
+#include <zephyr/logging/log.h>
+#include "bc12_pi3usb9201.h"
+
+LOG_MODULE_REGISTER(PI3USB9201, CONFIG_USB_BC12_LOG_LEVEL);
+
+/* Constant configuration data */
+struct pi3usb9201_config {
+	struct i2c_dt_spec i2c;
+
+	struct gpio_dt_spec intb_gpio;
+
+	enum bc12_type charging_mode;
+};
+
+/* Run-time configuration data */
+struct pi3usb9201_data {
+	const struct device *dev;
+	struct k_work work;
+	enum bc12_type partner_type;
+	struct gpio_callback gpio_cb;
+
+	bc12_callback_t result_cb;
+	void *result_cb_data;
+};
+
+enum pi3usb9201_client_sts {
+	CHG_OTHER = 0,
+	CHG_2_4A = 1,
+	CHG_2_0A = 2,
+	CHG_1_0A = 3,
+	CHG_RESERVED = 4,
+	CHG_CDP = 5,
+	CHG_SDP = 6,
+	CHG_DCP = 7,
+};
+
+struct bc12_status {
+	enum bc12_type partner_type;
+	int current_limit;
+};
+
+/*
+ * The USB Type-C specification limits the maximum amount of current from BC 1.2
+ * suppliers to 1.5A.  Technically, proprietary methods are not allowed, but we
+ * will continue to allow those.
+ */
+static const struct bc12_status bc12_chg_limits[] = {
+	[CHG_OTHER] = {BC12_TYPE_PROPRIETARY, 500},
```

Comment:
Why 500? Isn't the current limit in uA?

Reply:
Thanks, I fixed this.  I also updated to return the Isusp current limit to match a standard downstream port since the charger capabilities are unknown if the PI3USB9201 sets this bit.